geopandas/geoseries.py

```diff
@@ -61,16 +54,26 @@ def __init__(self, arg=None, index=None, crs=None, *args, **kwargs):
                 name = kwargs.get('name', None)
             else:
                 s = pd.Series(arg, index=index, **kwargs)
-                arg = from_shapely(s.values)
+                try:
+                    arg = from_shapely(s.values)
+                except TypeError:
+                    return s
```

Comment:
Maybe call this before doing the try-except?

```python
if s.dtype != object:
    return s
```

Reply:
good idea